debezium-connector-sqlserver/src/main/java/io/debezium/connector/sqlserver/SqlServerConnector.java

```diff
@@ -56,4 +66,49 @@ public void stop() {
     public ConfigDef config() {
         return SqlServerConnectorConfig.configDef();
     }
+
+    @Override
+    public Config validate(Map<String, String> connectorConfigs) {
+        Configuration config = Configuration.from(connectorConfigs);
+        SqlServerConnectorConfig sqlServerConfig  = new SqlServerConnectorConfig(config);
+        // First, validate all of the individual fields, which is easy since don't make any of the fields invisible ...
+        Map<String, ConfigValue> results = config.validate(SqlServerConnectorConfig.ALL_FIELDS);
+
+        // Get the config values for each of the connection-related fields ...
+        ConfigValue hostnameValue = results.get(SqlServerConnectorConfig.HOSTNAME.name());
+        ConfigValue portValue = results.get(SqlServerConnectorConfig.PORT.name());
+        ConfigValue databaseValue = results.get(SqlServerConnectorConfig.DATABASE_NAME.name());
+        ConfigValue userValue = results.get(SqlServerConnectorConfig.USER.name());
+        ConfigValue passwordValue = results.get(SqlServerConnectorConfig.PASSWORD.name());
+        final String passwordStringValue = config.getString(SqlServerConnectorConfig.PASSWORD);
+
+        if (Strings.isNullOrEmpty(passwordStringValue)) {
+            logger.warn("The connection password is empty");
+        }
+
+        // If there are no errors on any of these ...
+        if (hostnameValue.errorMessages().isEmpty()
+                && portValue.errorMessages().isEmpty()
+                && userValue.errorMessages().isEmpty()
+                && passwordValue.errorMessages().isEmpty()
+                && databaseValue.errorMessages().isEmpty()) {
+            // Try to connect to the database ...
+            try (SqlServerConnection connection = new SqlServerConnection(sqlServerConfig.jdbcConfig(), Clock.system(),
+                    sqlServerConfig.getSourceTimestampMode(), null)) {
+                // SqlServerConnection will try retrieving database, no need to run another query.
+                logger.info("Successfully tested connection for {} with user '{}'", connection.connectionString(),
+                        connection.username());
+            }
+            catch (Throwable e) {
+                logger.info("Failed testing connection for {} with user '{}'", sqlServerConfig.jdbcConfig(),
+                        userValue);
```

Comment:
Really, we should add an error about inability to connect (or inability to find tables) on all config associated properties in the `Config` object that we return. There's no trivial way to do this, but the errors in the `Config` are the **only** way to surface validation failures.

For example, if we can't connect using the hostname, port, user, password and database, then we should add an error like `Unable to connect. Check this and other connection properties.` to the hostname, port, user, password, and database config values.

Or, if we can connect but are unable to see at least one table matching the include & exclude filters in the config, we might add an error to those include and exclude filter config values.

Reply:
> to the hostname, port, user, password, and database config values.

@rhauch, do you actually mean to all of them? I'd say one, probably hostname, should suffice? It's one error after all, and we cannot tell which of the properties is causing it, but showing it more than once seems not be of any additonal help for the user?

Reply:
> are unable to see at least one table matching the include & exclude filters in the config, we might add an error to those include and exclude filter config values.

That's a tricky one; we already do _log_ a warning in this case. I don't think we should flag this as an error though, as for instance the user may set up the connector for an empty database, planning to create a matching table subsequently. Unfortunately, there's no means of raising a warning for a configuration value instead of a hard error.